packages/dbml-core/src/parse/error.js

```diff
@@ -0,0 +1,23 @@
+export class CompilerError {
+  /**
+   * @param {import("../../types").CompilerDiagnostic[]} diags
+   */
+  constructor (diags) {
+    this.diags = diags;
+  }
+
+  static create (nestedDiags) {
+    // eslint-disable-next-line no-use-before-define
+    return new CompilerError(flattenDiag(nestedDiags));
+  }
+
+  map (callback) {
+    return CompilerError.create(this.diags.map(callback));
```

Comment:
because the agruments is not nested here, you need to use .create to flatten them
```suggestion
    return new CompilerError(this.diags.map(callback));
```

Reply:
@NQPhuc  the old code is more generic anh, it allow `callback` to maps to arrays of diags, a CompilerError or simply a diag